storage/fuzz/fuzz_targets/ordinal_operations.rs

```diff
@@ -0,0 +1,322 @@
+                            let value = FixedBytes::new(value);
+
+                            if ordinal.put(index, value.clone()).await.is_ok() {
+                                expected_data.insert(index, value);
+                            }
+                        }
+
+                        // Sync after large batch to test buffer flushing
+                        if count > 0 && ordinal.sync().await.is_ok() {
+                            synced_data = expected_data.clone();
+                        }
+                    }
+                }
+
+                OrdinalOperation::ReopenAfterOperations => {
+                    if let Some(o) = store.take() {
+                        // Close the current ordinal (which includes a sync)
+                        o.close().await.expect("failed to close store before reopen failed");
+
+                        // Note: close() calls sync() internally, so update synced_data
+                        synced_data = expected_data.clone();
+
+                        // Reopen and verify synced data persisted
+                        match Ordinal::<_, FixedBytes<32>>::init(context.clone(), cfg.clone()).await
+                        {
+                            Ok(new_ordinal) => {
+                                // Verify all synced data is still accessible
+                                for (&index, expected_value) in synced_data.iter() {
+                                    match new_ordinal.get(index).await {
+                                        Ok(Some(value)) => {
+                                            assert_eq!(
+                                                &value, expected_value,
+                                                "Value at index {index} doesn't match after reopen",
+                                            );
+                                        }
+                                        Ok(None) => {
+                                            panic!(
+                                                "Synced value at index {index} missing after reopen",
+                                            );
+                                        }
+                                        Err(e) => {
+                                            panic!("Synced value at index {index} doesn't match after reopen: {e:?}");
+                                        }
+                                    }
+                                }
+
+                                // Continue with the new ordinal
+                                store = Some(new_ordinal);
+                                // Expected data remains the same after reopen
+                            }
+                            Err(e) => {
+                                panic!("Failed to reopen ordinal: {e:?}");
+                            }
+                        }
+                    }
+                }
+
+                OrdinalOperation::PruneAndPutInPrunedRange {
```

Comment:
nit: still pondering whether/how to enforce this behavior (the variable journal has a "prune guard" but not sure if we should replicate that pattern everywhere yet)